### PR TITLE
Done with the relationships.

### PR DESCRIPTION
**Pull request by bmakau**
# Description

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Change Status

- [x] Complete, tested, ready to review and merge
- [ ] Complete, but not tested (may need new tests)
- [ ] Incomplete/work-in-progress, PR is for discussion/feedback

# How Has This Been Tested?

- [ ] Test A
- [ ] Test B

# Checklist

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] My code has been reviewed by at least one peer
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] There are no merge conflicts

### CODE REVIEW

File: src/main/java/com/lambdaschool/wellness/model/Group.java [hunk 1]
```diff
@@ -3,30 +3,30 @@
...
 import javax.persistence.*;
-import java.util.ArrayList;
-import java.util.List;
-import java.util.Random;
+import java.util.*;
 
-// @Table(name = "groups") throws error when getting an user after creation, because groups is a reserved word.
 @Entity
-@Table(name = "groups_")
+@Table(name = "groups")
```
Comment (ckirkwood94) [inline src/main/java/com/lambdaschool/wellness/model/Group.java]: Groups is a reserved word. Need to change table name to something else.

File: src/main/java/com/lambdaschool/wellness/model/User.java [hunk 2]
```diff
@@ -29,33 +30,69 @@
 
     @JsonIgnore
     private String password;
+
+    @ManyToMany
+    @JoinTable(name = "user_group_competition",
+            joinColumns = {@JoinColumn(name = "userid")},
```
Comment (ckirkwood94) [inline src/main/java/com/lambdaschool/wellness/model/User.java]: joinColumns needs to reference "id" column on user table. `@JoinColumn(name = "userid", referencedColumnName = "id")`

File: src/main/java/com/lambdaschool/wellness/model/User.java [hunk 3]
```diff
@@ -29,33 +30,69 @@
...
+    inverseJoinColumns = {@JoinColumn(name = "groupid")})
+    private Set<Group> group = new HashSet<Group>();
+
+    @ManyToMany
+    @JoinTable(name = "user_group_competition",
+            joinColumns = {@JoinColumn(name = "userid")},
```
Comment (ckirkwood94) [inline src/main/java/com/lambdaschool/wellness/model/User.java]: joinColumns needs to reference "id" column on user table. `@JoinColumn(name = "userid", referencedColumnName = "id")`

File: src/main/java/com/lambdaschool/wellness/controllers/GroupController.java [hunk 4]
```diff
@@ -16,7 +16,7 @@
 
 @RestController
 @RequestMapping(value = "/api/group")
-@CrossOrigin("http:localhost:3000")
+@CrossOrigin(origins = "http://localhost:3000", maxAge = 3600)
```
Comment (Skidragon) [inline src/main/java/com/lambdaschool/wellness/controllers/GroupController.java]: Does this need to be here? You can do the CrossOrigin configuration in the WebConfig.java
In Intellij you can set environment variables in the program itself.

File: src/main/java/com/lambdaschool/wellness/model/Group.java [hunk 5]
```diff
@@ -3,30 +3,30 @@
...
 import javax.persistence.*;
-import java.util.ArrayList;
-import java.util.List;
-import java.util.Random;
+import java.util.*;
 
-// @Table(name = "groups") throws error when getting an user after creation, because groups is a reserved word.
 @Entity
-@Table(name = "groups_")
+@Table(name = "wellnes")
```
Comment (Skidragon) [inline src/main/java/com/lambdaschool/wellness/model/Group.java]: Would wellness_group be a better table name?

**Pull request merged**